src/test/java/com/gooddata/AbstractGoodDataAT.java

```diff
@@ -14,23 +14,36 @@
  */
 public abstract class AbstractGoodDataAT {
 
-    protected final String title;
-    protected final GoodData gd;
-
-    protected Project project;
-
-    protected String fact;
-    protected Attribute attr;
-    protected Metric metric;
-    protected Report report;
-    protected ReportDefinition reportDefinition;
-    protected ScheduledMail scheduledMail;
+    protected static final String title =
+            "sdktest " + new LocalDate() + " " + System.getenv("BUILD_NUMBER");
+
+    protected static final GoodData gd =
+            new GoodData(getProperty("host"), getProperty("login"), getProperty("pass"));
+
+    // tear down project and gd on shutdown
+    // this logic can't be in @After* methods, because we have inter-classes dependent tests
+    static {
+        Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
+            @Override
+            public void run() {
+                if (gd != null) {
+                    if (project != null) {
+                        gd.getProjectService().removeProject(project);
```

Comment:
I know this is ugly, alternatively we can define test group for all project dependent and setup `@AfterGroup` for project removal. However I see no better solution for `gd.logout()` :(